tmk_core/common/chibios/eeprom_stm32.h

```diff
@@ -47,6 +49,9 @@
 #    elif defined(MCU_STM32F103ZE) || defined(MCU_STM32F103RE) || defined(MCU_STM32F103RD) || defined(MCU_STM32F303CC) || defined(MCU_STM32F072CB)
 #        define FEE_PAGE_SIZE (uint16_t)0x800  // Page size = 2KByte
 #        define FEE_DENSITY_PAGES 4            // How many pages are used
+#    elif defined(MCU_STM32F411xE)
```

Comment:
Some docs explaining why you had to add a new flash driver rather than re-use the existing one would also be helpful for review.

Reply:
the explanation was added on the eeprom_stm32f4.c

Reply:
I would rather use the existing code if possible, and target an "improved" version across supported chips, instead of fragmenting implementations.

Reply:
stm32F4 has different register bits for flash programming compared with stm32f1 and f3.  It will introduce many hardware specific #define on the existing code. The new structure introduced by #7274 will not conflict the current EEPROM emulation implementation.

Reply:
I am just putting it on record that:
1. I'm not keen on having a non-uniform approach to the eeprom emulation strategy
2. I'm not keen on the code duplication this PR introduces